src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Sse41.PlatformNotSupported.cs

```diff
@@ -11,9 +11,9 @@ namespace System.Runtime.Intrinsics.X86
     /// This class provides access to Intel SSE4.1 hardware instructions via intrinsics
     /// </summary>
     [CLSCompliant(false)]
-    public static class Sse41
+    public abstract class Sse41 : Ssse3
```

Comment:
Here's the place where things get a little hairy with inheritance, right?

`Sse4.1` needs to check for both `Sse3` and `Ssse3`.  But `Ssse3` and `Sse3` have no direct releationship.  So really, we would need multiple inheritance here to correctly model this hierarchy.

Reply:
`Ssse3` has a direct dependency on `Sse3` and this was confirmed with Intel. This is a case where the manual is not self-consistent (12.7.1 lists the Sse3 dependency, even though 12.7.2 does not).

The only case where multiple inheritance is currently needed is for AVX-512F (which has not been reviewed/etc). Because of a dependency on both AVX2 and FMA, where they are siblings (both inherit from AVX). -- I called this bit out here: https://github.com/dotnet/coreclr/pull/19186#issuecomment-408620939

Reply:
> Ssse3 has a direct dependency on Sse3

I'm not seeing that modeled here:

https://github.com/tannergooding/coreclr/blob/001dddf2ad3091ba359bddc4f823d07c2e03573b/src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Ssse3.PlatformNotSupported.cs#L14

```C#
    public abstract class Ssse3 : Sse2
    {
```

Reply:
Mistake on my end. Fixed. -- I also went through the other ISAs and ensures they looked correct.